src/main/java/com/cincialert/controllers/UserController.java

```diff
@@ -1,53 +1,94 @@
 package com.cincialert.controllers;
 
+import com.cincialert.dao.UserDAO;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import com.cincialert.model.User;
+import com.cincialert.dto.User;
 import com.cincialert.service.UserService;
 
+import java.io.IOException;
 import java.util.List;
 
 @RestController
+
+
 @RequestMapping("/users")
 public class UserController {
 
     @Autowired
     private UserService userService;
 
+
+
     @PostMapping
     public ResponseEntity<User> createUser(@RequestBody User user) {
-        return ResponseEntity.ok(userService.createUser(user));
+        User newUser = userService.save(user);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        return new ResponseEntity<>(newUser, headers, HttpStatus.OK);
     }
 
     @GetMapping("/{id}")
     public ResponseEntity<User> getUserById(@PathVariable Long id) {
-        return userService.getUserById(id)
-                .map(ResponseEntity::ok)
-                .orElseGet(() -> ResponseEntity.notFound().build());
+        User user = userService.getUserById(id);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        return new ResponseEntity<>(user, headers, HttpStatus.OK);
     }
 
     @GetMapping
     public List<User> getAllUsers() {
-        return userService.getAllUsers();
+        return userService.findAll();
     }
 
     @PutMapping("/{id}")
     public ResponseEntity<User> updateUser(@PathVariable Long id, @RequestBody User user) {
-        if (!userService.getUserById(id).isPresent()) {
-            return ResponseEntity.notFound().build();
+        User updatedUser;
```

Comment:
Added error handling to most of your requests. Need this handling to see what issues have happened and prevent bugs.